model/parVarSheet.cc

```diff
@@ -40,6 +40,41 @@ namespace minsky
                                          return false;
                                        });   	
   }
+  
+  int ParVarSheet::colX(double x) const
+  { 
+	if (itemVector.empty()) return -1;
+    if ((x-offsx)<colLeftMargin[0]) return -1;
+    if ((x-offsx)<colLeftMargin[1]) return 0;
+    auto p=std::upper_bound(colLeftMargin.begin(), colLeftMargin.end(), (x-offsx));
+    size_t r=p-colLeftMargin.begin()-2+scrollColStart;
+    //if (r>cols()-1) r=-1; // out of bounds, invalidate. Also exclude A-L-E column. For ticket 1163.
+    if (r<0) r=-1;
+    return r;
+  }
+
+  int ParVarSheet::rowY(double y) const
+  {
+	if (itemVector.empty()) return -1;      
+    int c=(y-offsy)/(2*rowHeight);
+    //if (c>0) c+=scrollRowStart-1; 
+    //if (c<0 || size_t(c)>rows()) c=-1; // out of bounds, invalidate
+    if (c<0) c=-1; // out of bounds, invalidate
+    return c;
+  }
+  
+  ParVarSheet::ClickType ParVarSheet::clickType(double x, double y) const
+  {
+    int c=colX(x), r=rowY(y);
+        
+    //if (c>=0 && c<int(cols()))
+    //  if (r>=0 && r<int(rows()))
+    if (c>=0 && c<varAttribVals.size())
+      if (r>=0 && r<2*itemVector.size())
```

Comment:
the two conditionals can be combined by &&

Reply:
Done :)